docs/source/building/building.rst

```diff
@@ -165,7 +165,8 @@ or by providing arguments to the CMake call
  ``HiPACE_amrex_branch``       ``development``                           Repository branch for ``HiPACE_amrex_repo``
  ``HiPACE_amrex_internal``     **ON**/OFF                                Needs a pre-installed AMReX library if set to ``OFF``
  ``HiPACE_OPENPMD``            **ON**/OFF                                openPMD I/O (HDF5, ADIOS2)
-=============================  ========================================  =====================================================
+ ``HiPACE_PUSHER``             **AB5**/LEAPFROG                          Use fifth-order Adams-Bashforth or leapfrog Plasma Pusher
```

Comment:
@AlexanderSinn is it work making this a compile-time option?

I think pre-compiling both options and doing a runtime switch before the two specialized kernels would be more user friendly.

Reply:
Discussed: needs additional (25) force terms on the particle, so that's a bit of a heavy lift.

Reply:
As mentioned, it is a compile-time option not for performance of the push kernel, but rather to save on memory of the 25 Adams-Bashforth force terms (per plasma particle) if the leapfrog pusher is used. One would need to change these to runtime components to make this a runtime option.